src/commands/install/pull_package_submodules_test.py

```diff
@@ -0,0 +1,135 @@
+import os
+from os import mkdir, path
+from unittest.mock import MagicMock
+
+import pytest
+from git.repo import Repo
+from pytest_mock import MockerFixture
+
+from src.commands.install.pull_package_submodules import pull_package_submodules
+
+# - repo
+#   - lib
+#     - package_repo (s)
+# - repo_clone
+#   - ...
+# - package_repo
+#   - foo.txt
+
+
+@pytest.fixture(name="packages_dir_name")
+def fixture_packages_dir_name() -> str:
+    return "lib"
+
+
+@pytest.fixture(name="the_packages_file_name")
+def fixture_the_packages_file_name() -> str:
+    return "foo.txt"
+
+
+@pytest.fixture(name="the_package_name")
+def fixture_the_package_name() -> str:
+    return "package"
+
+
+@pytest.fixture(name="repo_dir")
+def fixture_repo_dir(tmpdir: str) -> str:
+    return path.join(tmpdir, "repo")
+
+
+@pytest.fixture(name="repo_clone_dir")
+def fixture_repo_clone_dir(tmpdir: str) -> str:
+    return path.join(tmpdir, "repo_clone")
+
+
+@pytest.fixture(name="package_repo_dir")
+def fixture_package_repo_dir(tmpdir: str) -> str:
+    return path.join(tmpdir, "package_repo")
+
+
+@pytest.fixture(name="package_repo")
+def fixture_package_repo(package_repo_dir: str, the_packages_file_name: str) -> Repo:
+    repo = Repo.init(package_repo_dir)
+
+    the_file_path = path.join(package_repo_dir, the_packages_file_name)
+    with open(path.join(the_file_path), "w", encoding="utf-8") as file:
+        file.write("foo")
+
+    repo.git.add("-A")
+    repo.index.commit("add foo")
+
+    assert path.exists(the_file_path)
+
+    return repo
+
+
+@pytest.fixture(name="repo")
+# pylint: disable=unused-argument
+def fixture_repo(
+    repo_dir: str,
+    package_repo_dir: str,
+    packages_dir_name: str,
+    the_package_name: str,
+    package_repo: Repo,
+) -> Repo:
+    repo = Repo.init(repo_dir)
+
+    packages_dir = path.join(repo_dir, packages_dir_name)
+    package_dir = path.join(packages_dir, the_package_name)
+    mkdir(packages_dir)
+    repo.create_submodule(the_package_name, package_dir, package_repo_dir)
+
+    repo.git.add()
+    repo.index.commit("add submodule")
+
+    return repo
+
+
+@pytest.fixture(name="repo_clone")
+# pylint: disable=unused-argument
+def fixture_repo_clone(
+    repo_clone_dir: str, repo_dir: str, repo: Repo, packages_dir_name: str
+) -> Repo:
+    cloned_repo = repo.clone(repo_clone_dir)
+
+    assert path.exists(
+        path.join(repo_clone_dir, packages_dir_name)
+    ), "./lib exists in the cloned repo directory"
+    return cloned_repo
+
+
+# ----------------------------------- TESTS ---------------------------------- #
```

Comment:
> Please, do not use things like this. Such logical separation is purpose why modules exiss, please create fixtures.py and put all of the fixtures for install module there.
I'll do it. The reason I did what I do was consistency. Currently, all "private" fixtures are kept in the unit test files and “comment dividers” are commonly used in Swift.

Reply:
- removed comment dividers
- didn't extracted fixtures to separate file

According to pytest docs fixtures should be kept in `conftest.py` however this causes fixtures to be shared across tests in the module and its submodules and there's no point of create module for 1 file.
`pull_package_submodules` require special environment (`repo_clone` directory) because when a submodule is added, git fetches its content, and it cannot be disabled.